tests/MatchesSnapshotTest.php

```diff
@@ -0,0 +1,223 @@
+<?php
+
+namespace Spatie\Snapshots\Test;
+
+use Spatie\Snapshots\MatchesSnapshots;
+use PHPUnit_Framework_MockObject_MockObject;
+use PHPUnit_Framework_ExpectationFailedException;
+
+class MatchesSnapshotTest extends TestCase
+{
+    /** @test */
+    public function it_can_create_a_snapshot_from_a_string()
+    {
+        $mockTrait = $this->getMatchesSnapshotMock();
+
+        $this->expectIncompleteMatchesSnapshotTest($mockTrait);
+
+        $mockTrait->assertMatchesSnapshot('Foo');
+
+        $snapshot = $this->snapshotsDir.DIRECTORY_SEPARATOR.'MatchesSnapshotTest__it_can_match_an_existing_string_snapshot.php';
```

Comment:
Could we pull out all the filesystem methods & properties to a `Filesystem` class or something? Would be nice to not have to stare at `DIRECTORY_SEPERATOR` calls in the test methods. Maybe we could even add assertion method on the filesystem? Like `$this->filesystem->assertSnapshotEquals('MatchesSnapshotTest__it_can_match_an_existing_string_snapshot.php', "<?php return 'Foo';")`.